spec/src/language_negotiation_lambda_spec.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+var languageNegotiationLambda = require( '../../src/language_negotiation_lambda' );
+
+describe( 'languageNegotiationLambda', function() {
+    var event,
+        default_header = ['en'];
+
+    beforeEach(function(){
+        // Cloudfront Request Object
+        event = {
+            "Records": [
+                {
+                  "cf": {
+                    "request": {
+                      "headers": {
+                        "Host": [
+                          "localhost"
+                        ],
+                        "User-Agent": [
+                          "Test Agent"
+                        ]
+                      },
+                      "clientIp": "2001:cdba::3257:9652",
+                      "httpVersion": "2.0",
+                      "uri": "/",
+                      "method": "GET"
+                    },
+                    "config": {
+                      "distributionId": "EXAMPLE"
+                    }
+                  }
+                }
+            ]
+        };
+    });
+
+    describe( 'Valid cookies', function(){
+        [
+            { cookie: "en",     expected: "en" },
+            { cookie: "EN",     expected: "en" },
+            { cookie: "es",     expected: "es" },
+            { cookie: "ES",     expected: "es" },
+            { cookie: "en-US",  expected: "en" },
+            { cookie: "es-419", expected: "es" },
+            { cookie: "es_419", expected: "es" },
+            { cookie: "fr",     expected: "es" },
+            { cookie: "",       expected: "es" }
+        ].forEach( function( cookies ) {
+            it( `Resolves cookie ${cookies.cookie} to ${cookies.expected}`, function( done ) {
+                event.Records[0].cf.request.headers.Cookie = [
+                    process.env.LANGUAGE_COOKIE_NAME+"="+cookies.cookie
+                ];
+                event.Records[0].cf.request.headers['Accept-Language'] = ['es'];
```

Comment:
It might be better for this to be something random, like 'eo' so that we can be sure the test cases resolve to `es` because the cookie logic is correct, not because it fell back to Accept-Language.